api/src/services/listing.service.ts

```diff
@@ -1062,6 +1058,47 @@ export class ListingService implements OnModuleInit {
     return undefined;
   }
 
+  /**
+   * @param listingId the listing id we are operating on
+   * @description disconnects assets from listing events, then deletes those assets
+   */
+  async updateListingEvents(listingId: string): Promise<void> {
+    const assetIds = await this.prisma.listingEvents.findMany({
+      select: {
+        id: true,
+        fileId: true,
+      },
+      where: {
+        listingId,
+      },
+    });
+    await Promise.all(
+      assetIds.map(async (assetData) => {
+        await this.prisma.listingEvents.update({
+          data: {
+            assets: {
+              disconnect: true,
+            },
+          },
+          where: {
+            id: assetData.id,
+          },
+        });
+      }),
+    );
+    await Promise.all(
+      assetIds.map(async (assetData) => {
+        if (assetData.fileId) {
+          await this.prisma.assets.delete({
+            where: {
+              id: assetData.fileId,
+            },
+          });
+        }
+      }),
```

Comment:
Could this be done as a `deleteMany` where id is in assetId list?

Reply:
nice call out! updated!